File: app/src/main/java/com/niu/protect/manager/AutoSettingManager.java
package com.niu.protect.manager;

import android.content.Context;
import com.niu.protect.tools.ILog;
public class AutoSettingManager {
    public static final int AUTO_SETTING_AUTOSETTING = 2;
    public static final int AUTO_SETTING_BYHAND = 1;
    public static final int AUTO_SETTING_FINISH = 3;
    public static final int AUTO_SETTING_NOT_SET = 0;
    private static AutoSettingManager instance;
    private int autoSettingStatus = 0;

    public static AutoSettingManager getInstance() {
        if (instance == null) {
            synchronized (AutoSettingManager.class) {
                if (instance == null) {
                    instance = new AutoSettingManager();
                }
            }
        }
        return instance;
    }

    public int getAutoSettingStatus() {
        return this.autoSettingStatus;
    }

    public void setAutoSettingFinish(Context context, int isFinish) {
        this.autoSettingStatus = isFinish;
        SharedPreManager.saveAutoSettingFinish(context, isFinish);
    }

    public int getAutoSettingFinish() {
        if (this.autoSettingStatus == 0) {
            this.autoSettingStatus = SharedPreManager.getAutoSettingFinish();
        }
        return this.autoSettingStatus;
    }

    public boolean isNeedAutoSetting() {
        getAutoSettingFinish();
        int i = this.autoSettingStatus;
        return (i == 3 || i == 1) ? false : true;
    }

    public synchronized boolean isSettingFinish() {
        getAutoSettingFinish();
        ILog.d("auto set", "------" + this.autoSettingStatus);
        return this.autoSettingStatus == 3;
    }
}

File: protect/src/main/java/com/niu/protect/manager/AutoSettingManager.kt
package com.niu.protect.manager

import com.niu.protect.manager.SharedPreManager.saveAutoSettingFinish
import com.niu.protect.tools.ILog

object AutoSettingManager {
    const val AUTO_SETTING_NOT_SET = 0
    const val AUTO_SETTING_HAND = 1
    const val AUTO_SETTING_AUTO = 2
    const val AUTO_SETTING_FINISH = 3
//    private val instance: AutoSettingManager? = null
    var autoSettingFinish: Int
        get() = SharedPreManager.autoSettingFinish
        set(step) {
            saveAutoSettingFinish(step)
        }
    val isNeedAutoSetting: Boolean
        get() {
            val i = autoSettingFinish
            return i != AUTO_SETTING_FINISH && i != AUTO_SETTING_HAND
        }

    @get:Synchronized
    val isSettingFinish: Boolean
        get() {
            val i = autoSettingFinish
            ILog.d("auto set", "------$i")
            return i == AUTO_SETTING_FINISH
        }
}